content/developer/howtos/upgrade_custom_db.rst

```diff
@@ -0,0 +1,281 @@
+database. To request the upgraded test database, follow the steps mentioned in
+:ref:`upgrade/request-test-database`, depending on the hosting type of your database.
+
+The purpose of this stage is not to start working with the custom modules in the upgraded database,
+but to make sure the standard upgrade process works seamlessly, and the test database is delivered
+properly. If that's not the case, and the upgrade request fails, request the assistance of Odoo via
+the `support page <https://odoo.com/help?stage=migration>`_ by selecting the option related to
+testing the upgrade. 
+
+
+.. _upgrade_custom/empty_database:
+
+Empty database
+==============
+
+Before working on an upgraded test database, we recommend to make the custom developments work on an
+empty database in the targeted version of your upgrade. 
+This ensures that the customization is compatible with the new version of Odoo, allows to analyse
+how does it behave and interact with the new features, and guarantees that they will not cause any
+issue when upgrading the database.
+
+Making the custom modules work in an empty database also helps avoiding changes and wrong
+configurations that might be present on the production database (like studio customization,
+customized website pages, email templates or translations). They are not intrinsically related to
+the custom modules and that can raise unwanted issues in this stage of the upgraded process.
+
+To make custom modules work on an empty database we advise to follow these steps:
+
+- :ref:`upgrade_custom/empty_database/modules_installable`
+- :ref:`upgrade_custom/empty_database/test_fixes`
+- :ref:`upgrade_custom/empty_database/clean_code`
+- :ref:`Make standard tests run successfully <upgrade_custom/empty_database/standard_test>`
+
+.. _upgrade_custom/empty_database/modules_installable:
+
+Make custom modules installable
+-------------------------------
+
+The first step is to make the custom modules installable in the new Odoo version.
+This means, starting by ensuring there are no tracebacks or warnings during their installation.
+For this, install the custom modules, one by one, in an empty database of the new Odoo version and
+fix the tracebacks and warnings that arise from that.
+
+.. todo:: Add examples or references to PR such as attrs change
+
+This process will help detect issues during the installation of the modules. For example:
+
+- Invalid module dependencies.
+- Syntax change: assets declaration, OWL updates, attrs.
+- References to standard fields, models, views not existing anymore or renamed.
+- Xpath that moved or were removed from views.
+- Methods renamed or removed.
+- ...
+
+.. _upgrade_custom/empty_database/test_fixes:
+
+Test and fixes
+--------------
+
+Once there are no more tracebacks when installing the modules, the next step is to test them.
+Even if the custom modules are installable on an empty database, this does not warranty there are no
```

Comment:
I think this should be `this does not warrant ...` as warranty is the common word, while warrant is the verb :D